test/StaticValidatorTest.php

```diff
@@ -152,9 +152,38 @@ public function testPassingNullWhenSettingPluginManagerResetsPluginManager()
         $this->assertNotSame($plugins, StaticValidator::getPluginManager());
     }
 
-    public function testExecuteValidWithParameters()
+    public function parameterizedData()
     {
-        $this->assertTrue(StaticValidator::execute(5, 'Between', [1, 10]));
-        $this->assertTrue(StaticValidator::execute(5, 'Between', ['min' => 1, 'max' => 10]));
+        return [
+            'valid-positive-range'   => [5, 'Between', ['min' => 1, 'max' => 10], 'assertTrue'],
+            'valid-negative-range'   => [-5, 'Between', ['min' => -10, 'max' => -1], 'assertTrue'],
+            'invalid-positive-range' => [-5, 'Between', ['min' => 1, 'max' => 10], 'assertFalse'],
+            'invalid-negative-range' => [5, 'Between', ['min' => -10, 'max' => -1], 'assertFalse'],
+        ];
+    }
+
+    /**
+     * @dataProvider parameterizedData
+     */
+    public function testExecuteValidWithParameters($value, $validator, $options, $assertion)
+    {
+        $this->$assertion(StaticValidator::execute($value, $validator, $options));
```

Comment:
The contract for validators is to return boolean. Adding the assertion on
type is superfluous.

On Wed, Jul 22, 2015 at 8:01 AM, Maks3w <notifications@github.com> wrote:

> In test/StaticValidatorTest.php
> <https://github.com/zendframework/zend-validator/pull/26#discussion_r35209413>
> :
>
> > -        $this->assertTrue(StaticValidator::execute(5, 'Between', [1, 10]));
> > -        $this->assertTrue(StaticValidator::execute(5, 'Between', ['min' => 1, 'max' => 10]));
> > +        return [
> > +            'valid-positive-range'   => [5, 'Between', ['min' => 1, 'max' => 10], 'assertTrue'],
> > +            'valid-negative-range'   => [-5, 'Between', ['min' => -10, 'max' => -1], 'assertTrue'],
> > +            'invalid-positive-range' => [-5, 'Between', ['min' => 1, 'max' => 10], 'assertFalse'],
> > +            'invalid-negative-range' => [5, 'Between', ['min' => -10, 'max' => -1], 'assertFalse'],
> > +        ];
> > +    }
> > +
> > +    /**
> > +     * @dataProvider parameterizedData
> > +     */
> > +    public function testExecuteValidWithParameters($value, $validator, $options, $assertion)
> > +    {
> > +        $this->$assertion(StaticValidator::execute($value, $validator, $options));
>
> The problem with assersame is anyone can put a string in the provider.
>
> Split in two make more clear 1:1 relations with method signature and avoid
> put false positive values in provider.
>
> This is motivated in cases where the provider is implemented in child
> classes while the test method is not
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-validator/pull/26/files#r35209413>.
>



-- 
Matthew Weier O'Phinney
matthew@weierophinney.net
https://mwop.net/